### ISSUE TITLE
regexp/remove-useless-group false positives

### ISSUE THREAD

**Issue by EvgenyOrekhov** [coderaiser/putout#45]
`regexp/remove-useless-group` produces too many false positives.

Example code from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges):

```js
const imageDescription = 'This image has a resolution of 1440×900 pixels.';
const regexpSize = /([0-9]+)×([0-9]+)/;
const match = imageDescription.match(regexpSize);
console.log(`Width: ${match[1]} / Height: ${match[2]}.`);
```

Putout recommends using `/[0-9]+×[0-9]+/`, but the `console.log()` line will not work in that case.

Another example (an [ember-cli-showdown](https://github.com/gcollazo/ember-cli-showdown) initializer):

```js
function initialize() {
  showdown.extension("smallLinks", function smallLinks() {
    return [
      {
        type: "output",
        regex: /<a (?<attributes>.*)>/u,
        replace: '<a class="link link--small" $<attributes>>',
      },
    ];
  });
}
```

Putout recommends using `/<a .*>/u`, but it will break the code (the `replace` will stop working properly).

Basically, there is no way to know whether capture groups will be used or not.

**Comment by EvgenyOrekhov** [comment]
Another false positive:

```js
/(.)\1{3,}/gi.test("abcccc"); // result: true
```

Putout recommends changing it to:

```js
/.\1{3,}/gi.test("abcccc"); // result: false
```

**Comment by coderaiser** [comment]
Thanks a lot for a bug report with so many examples, just fixed with https://github.com/coderaiser/putout/commit/a71e8152ef06a0e84a13662c982f3e1df801a7c4 🎉 .

Is it works for you?

**Comment by EvgenyOrekhov** [comment]
It works, thank you!
